honeybee/boundarycondition.py

```diff
@@ -21,18 +21,40 @@ def __init__(self, sun_exposure=False, wind_exposure=False,
                 This input can also be the word 'autocalculate' to have the view
                 factor automatically calculated.  Default: 'autocalculate'.
         """
+        sun_exposure = False if sun_exposure is None else sun_exposure
         assert isinstance(sun_exposure, bool), \
             'Input sun_exposure must be a Boolean. Got {}.'.format(type(sun_exposure))
         self._sun_exposure = sun_exposure
+
+        wind_exposure = False if wind_exposure is None else wind_exposure
         assert isinstance(wind_exposure, bool), \
             'Input wind_exposure must be a Boolean. Got {}.'.format(type(wind_exposure))
         self._wind_exposure = wind_exposure
-        if view_factor == 'autocalculate':
-            self._view_factor = view_factor
+
+        if view_factor == 'autocalculate' or view_factor is None:
+            self._view_factor = 'autocalculate'
         else:
             self._view_factor = float_in_range(
                 view_factor, 0.0, 1.0, 'view factor to ground')
 
+    @classmethod
+    def from_dict(cls, data):
+        """Initialize BoundaryCondition from a dictionary.
+
+        Args:
+            data: A dictionary representaion of the boundary condition.
+        """
+        if 'sun_exposure' in data:
+            sun_exposure = True if data['sun_exposure'] == 'SunExposed' else False
```

Comment:
you probably want to do the comparison for lower(). In EP `sunexposed` or `SUNExposed` are also valid.

Reply:
I just made a decision to have `sun_exposure` as a boolean in the schema.  This property is just much better described with a "yes or no" answer that the text values gets unnecessarily confusing, creating more room for errors than they eliminate.  However, I took your suggestion and used a `lower()` check for the `autocalculate` parameter of the `view_factor`.